### ISSUE TITLE
Place auto-generated JSDoc comments above multiple decorators/annotations 

### ISSUE THREAD

**Issue by david-sharer** [gajus/eslint-plugin-jsdoc#659]
When using TypeScript w/ multiple Annotations, eslint-plugin-jsdoc is expecting the doc to appear in a strange place. #571 indicates that the doc comment should be above annotations (and it seems to work for single annotations), but in the case of multiple annotations the behavior does not match.

## Expected behavior
The following code snippet does not trigger the rule jsdoc/require-jsdoc.
```ts
/** Defines the current user's settings. */
@Injectable({
  providedIn: 'root',
})
@State<Partial<UserSettingsStateModel>>
({
  name: 'userSettings',
  defaults: {
    isDev: !environment.production,
  },
})
export class UserSettingsState { }
```

## Actual behavior
The above code triggers the rule jsdoc/require-jsdoc, even though VS Code detects the documentation correctly.

It considers this valid:
```ts
@Injectable({
  providedIn: 'root',
})
@State<Partial<UserSettingsStateModel>>
({
  name: 'userSettings',
  defaults: {
    isDev: !environment.production,
  },
})
/** Defines the current user's settings. */
export class UserSettingsState { }
```

When run with --fix and no documentation, it attempts to place the doc comment line so:
(AFAIK this is invalid, and VS Code does not pick it up)
```ts
@Injectable({
  providedIn: 'root',
})
@State<Partial<UserSettingsStateModel>>/**
 *
 */
({
  name: 'userSettings',
  defaults: {
    isDev: !environment.production,
  },
})
export class UserSettingsState { }
```

## ESLint Config

```js
module.exports = {
  root: true,
  overrides: [
    {
      files: ['*.ts'],
      parser: '@typescript-eslint/parser',
      parserOptions: {
        project: [
          'tsconfig.*?.json',
        ],
        createDefaultProgram: true,
      },
      extends: [ ],
      plugins: ['jsdoc'],
      rules: {
        'jsdoc/require-jsdoc': [
          'error',
          {
            require: {
              ClassDeclaration: true,
            },
          },
        ],
      },
    },
  ],
};
```

## ESLint sample

```ts
import { Injectable } from '@angular/core';
import { environment } from '@environments/environment';
import { State } from '@ngxs/store';

/** Defines the user state model. */
export class UserSettingsStateModel {
  public isDev: boolean;
}

/** Contains user-settings state. */
@Injectable({
  providedIn: 'root',
})
@State<Partial<UserSettingsStateModel>>({
  name: 'userSettings',
  defaults: {
    isDev: !environment.production,
  },
})
export class UserSettingsState { }
```

## Environment

- Node version: v12.17.0
- ESLint version v7.13.0
- `eslint-plugin-jsdoc` version: 30.7.8
